### ISSUE TITLE
Two visual bugs.

### ISSUE THREAD

**Issue by HolySoap** [blackadderkate/weather-widget-2#17]
![Screenshot_2022-01-19_20:56:02](https://user-images.githubusercontent.com/65773051/150204760-f89c9fab-ab1f-4c6b-960a-a7c675506a4b.png)

The first one, that you can hopefully see, is the tiny wind speed indicator. When you hover with you mouse over the small wind direction symbol EVERY SECOND (the one between full hours) information becomes glitched.

The second one is the symbol on the Dock, its not centred.

**Comment by blackadderkate** [comment]
I am aware of the windspeed bug - it is to do with the way transparency works in QT. I will try to spend some time looking at this over the weekend. I am a noob at QT coding, and I need to do some research... 

I will have a look at the formatting of the widget in the dock - that should be an easier fix...

**Comment by HolySoap** [comment]
Take your time, I'm just happy that someone continues this project.

I probably should mention that the dock is Latte Dock (v0.10.7) and its running on Wayland.
There is another one I found and overlooked.

![Screenshot_2022-01-20_10:22:15](https://user-images.githubusercontent.com/65773051/150310642-fc3dfda5-33b0-4898-9735-e7be544e8628.png)

The text on the bottom left is almost unreadable on a dark Plasma theme. I don't know what the easier solution is, make it white for dark themes and black for light themes or make it show in the negative colour.

**Comment by blackadderkate** [comment]
That last one is a simple fix. I'll fix it when I get back from work.

**Comment by blackadderkate** [comment]
The code for sizing the widget in the dock makes no sense to me at all, and it does not seem to work properly. I think I will have to rewrite it. :-(
As a quick fix, I have centred the temperature (instead of it being right-aligned) - which I think looks a bit better. I'll upload a new build here - but I won't release it on the KDE store until I have fixed all the bugs on this report.

**Comment by HolySoap** [comment]
Have you considered asking for help, like in the [Qt Forum](https://forum.qt.io/)? 
I'm sure as hell someone has an easy and quick answer for this.

**Comment by blackadderkate** [comment]
FYI I've been working on the widget formatting issue today.
I created a new test widget with "dummy" data, that seems to format properly. Now I just need to transfer it over to the actual WeatherWidget, and see whether it works on the real widget.

For the Windspeed issue, I have found a "HintBox" QML component on Github which might fix the problem. When time allows I will give it a try.

**Comment by blackadderkate** [comment]
I have a fix for the WindSpeed indicator - but it requires QtQuick.Controls 2.5 (Qt 5.12 or greater).
Before I check it in, I just want to make sure you're not running something really old like Kubuntu 18.04LTS?

**Comment by HolySoap** [comment]
Do not worry, I'm running openSUSE Tumbleweed and do my updates daily.

**Comment by solbu** [comment]
> I have a fix for the WindSpeed indicator - but it requires QtQuick.Controls 2.5 (Qt 5.12 or greater). Before I check it in, I just want to make sure you're not running something really old like Kubuntu 18.04LTS?

I'm one of those who also look forward to using this plugin.
I use a slightly older distro than @HolySoap (Mageia 7, released in 2019), but it has Qt 5.12.6, so it should work on my system also – once I get around to actually testing it. ;-)

**Comment by blackadderkate** [comment]
Just checked the Windspeed fix into the placeSearch branch.
I think I've also fixed the widget formatting in the dock (it looks perfect on Kubuntu 20.04) and the black text in the Configuration Page - which I think is everything?

![Screenshot_20220125_181908](https://user-images.githubusercontent.com/83865293/151035883-2baa5285-51f0-48c0-9671-86237f11d8e8.png)

[Download the New Version here.](https://github.com/blackadderkate/weather-widget-2/releases/download/v2.1.5/weather-widget-2.1.5.plasmoid)

**Comment by HolySoap** [comment]
Attaching the wind speed to the mouse courser is a neat idea and you got rid of the hard to read white on red, very nice.
Yes these are the issues I noticed, but I wait for an update trough Discover and I'm very sorry for the inconvenience it brings to you regarding fast feedback.

I have a little side question that bothers me since the original plasmoid and maybe you have an answer. 
What logic is the Y-axis for degrees following? Sometimes you see even numbers and sometimes uneven numbers.

**Comment by blackadderkate** [comment]
Thanks for the feedback.

I knew when I first released this update that there would be a LOT of bug reports, simply because this is the first plasmoid I have ever written. There are so many different Linux distributions and KDE plasma versions, not to mention Themes, Languages, and Timezones - and I can only test a tiny fraction of those.

I had a feeling that someone who goes to the trouble of filing a bug report on Github would probably be quite technically minded, so I decided that I would try to put "bugfixes" on here first, and wait for the person who reported the bug to tell me whether it was fixed or not before publishing it on the KDE store / Discover.

And if I did publish a new version on the KDE store every time I fixed a "minor" bug, people might get the impression that the widget is literally full of bugs and give it a bad review, or uninstall it.

I hope that makes sense?


As for the logic, I literally have no idea!
When I look at someone else's code, I can usually understand how it works quite well - but not with this widget! Like a lot of code I think it started off quite small and easy to understand, and as new bits got added it just got more and more complicated...

I assume it uses whatever makes the curve sit in the middle of the grid - so the Y axis depends on what the Highest and Lowest temperatures are...
